Simplenote/Classes/ViewSpinner.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+import QuartzCore
+
+// MARK: - ViewSpinner: Spin a view
+//
+final class ViewSpinner {
+    private let view: UIView
+    private var displayLink: CADisplayLink?
+
+    private var angle: Double = 0.0
+    private var velocity: Double = 0.0
+
+    /// Callback is invoked once max velocity is reached
+    ///
+    var onMaxVelocity: (() -> Void)?
+
+    /// Init with a view to spin
+    ///
+    init(view: UIView) {
+        self.view = view
+    }
```

Comment:
I've noticed `stop()` isn't invoked on deinit. However, DisplayLink appears to be handling things well.

Not sure if we do need to invoke `invalidate()` anyways?

Reply:
`CADisplayLink` retains the target and there is a retain loop `ViewSpinner` <-> `CADisplayLink`. `CADisplayLink` releases the target when we call `stop` on `ViewSpinner`. This happens when max velocity is reached or when view controller disappears from the screen.
When `ViewSpinner` deinit is called, `CADisplayLink` should already be nil.